### ISSUE TITLE
Webapp: custom handler.py can't opt out of auto-detected criterion

### ISSUE THREAD

**Issue by fazelehh** [aidotse/LeakPro#435]
Follow-up from review on #423 ([comment](https://github.com/aidotse/LeakPro/pull/423#discussion_r3719098555)).

## Context

In `leakpro/webapp/backend/main.py`, the training step auto-picks a criterion based on the probed model output shape:

```python
if _is_binary:
    criterion = nn.BCEWithLogitsLoss()
else:
    criterion = nn.CrossEntropyLoss()
```

Today `TrainParams` has no `criterion` field at all, so for the normal (built-in) training path there's nothing to override — this is the only thing that ever sets `criterion`.

## The actual gap

The webapp supports uploading a custom `handler.py` (Step 3's custom training loop) with its own `train()`/`eval()` methods. That custom handler still receives this auto-detected `criterion` as an argument, force-passed in:

```python
result = _handler.train(loader, model, criterion, optimizer, epochs=params.epochs, **_dpsgd_kwargs)
```

A custom handler that wants to define or select its own loss internally (e.g. a weighted loss, focal loss, or a criterion tied to some app-specific config) has no way to opt out of the auto-detected one.

## Possible fix

Let a custom `handler.py` declare its own criterion (e.g. a class attribute or a `get_criterion()` method) and skip auto-detection when present, falling back to the current auto-detect behavior otherwise. Scoped to the custom-handler path only — no schema or frontend changes needed for the common case.
